Fix an issue when ESB_NEVER_DISABLE_TX is enabled.

LGrosbois-WTK opened the pull request:
The experimental feature `ESB_NEVER_DISABLE_TX` can achieve delay below 100µs in-between data transmission and reception. When input data flow in not fast enough, and ESB fifo gets empty, static variable `esb_state` remains to `ESB_STATE_PTX_TXIDLE` instead of `ESB_STATE_IDLE`. 
The esb_write_payload checks this flag before initiating a new transmission, and the new case where `esb_write_payload = ESB_STATE_PTX_TXIDLE` was forgotten.

This bugfix follows the experimental feature PR (now merged) : https://github.com/nrfconnect/sdk-nrf/pull/9835

NordicBuilder commented:
Thank you for your contribution!
It seems you are not a member of the nrfconnect GitHub organization. External contributions are handled as follows:
Large contributions, affecting multiple subsystems for example, may be rejected if they are complex, may introduce regressions due to lack of test coverage, or if they are not consistent with the architecture of nRF Connect SDK.
PRs will be run in our continuous integration (CI) test system.
If CI passes, PRs will be tagged for review and merged on successful completion of review. You may be asked to make some modifications to your contribution during review.
If CI fails, PRs may be rejected or may be tagged for review and rework.
PRs that become outdated due to other changes in the repository may be rejected or rework requested.
External contributions will be prioritized for review based on the relevance to current development efforts in nRF Connect SDK.  Bug fix PRs will be prioritized.
You may raise issues or ask for help from our Technical Support team by visiting https://devzone.nordicsemi.com/.

*Note: This comment is automatically posted and updated by the Contribs GitHub Action.*

LGrosbois-WTK commented:
Hello @lemrey !
Is it possible to add the flag so that the PR is run in the CI ?
As it is a light edit impacting only an experimental feature, it should be fast to review :)

Thanks !

LGrosbois-WTK commented:
Hi !
As the CI automatically added the changelog-entry-required flag, I updated the changelog.
Although, I do not know if it is really needed, as the fixed feature has just been introduced (i.e. is new in the changelog, and has not been released yet).

Please let me know if I need to remove it.

lemrey reviewed (approved):
Hello @LGrosbois-WTK , and thanks for fixing this up :) There's no need for a changelog entry, given that this issue affects only the `main` branch, and no NCS release is affected; I will remove the label.

As for the fix itself, @KAGA164 could you please have a look?

LGrosbois-WTK commented:
Hi, just removed the changelog additions :)

KAGA164 reviewed (approved):
Looks good to me

LGrosbois-WTK commented:
Thanks @KAGA164  for your review.
@lemrey : is it possible to add the CI flag so the code is tested and maybe merged ?

LGrosbois-WTK commented:
I @lemrey and @KAGA164 
All CI checks have passed, and the PR has 2 approvals.
Is it possible to merge or does this PR needs more reviews ?

KAGA164 commented:
> Hi ! A little follow-up on this issue. Is there anything I can do to help merging this bugfix ?

@LGrosbois-WTK Thanks for reminding. When the CI will be green, I will try to get it merged

LGrosbois-WTK commented:
The CI started 28 days ago...
Is there a way to reboot it ?

KAGA164 commented:
> The CI started 28 days ago... Is there a way to reboot it ?
> 
> Should I rebase ?

@LGrosbois-WTK Yes, please rebase

NordicBuilder commented:
You can find the documentation preview for this PR at this <a href='http://developer.nordicsemi.com/nRF_Connect_SDK_dev/doc/PR-10965
'>link</a>. It will be updated about 10 minutes after the documentation build succeeds.

*Note: This comment is automatically posted by the Documentation Publishing GitHub Action.*

LGrosbois-WTK commented:
I've just rebased.
Apparently a lot has changed in the CI...
There are some failed tests not related to what I have modified.
As they are in jenkins, I cannot access the logs.

Any inside information ?

LGrosbois-WTK commented:
@KAGA164  @lemrey Any news on the CI failed tests ?
